"failed to parse package patterns" from yarn.lock after upgrade to "aquasec/trivy:0.36.1"

## Description

After upgrade of my testing command to use `aquasec/trivy:0.36.1` I'm getting a `yarn.lock` parsing error if I enable debug mode…

## What did you expect to happen?

See the yarn package vulnerabilities

## What happened instead?

The `yarn.lock` is seemingly ignored without debug, otherwise tells that parsing failed.

## Output of run with `-debug`:

```
docker run --rm -it \
  --volume="/opt/dev/academy:/root/app/" \
  aquasec/trivy:0.36.1 fs "/root/app" \
    --skip-dirs="node_modules" \
    --config="/root/app/trivy.yaml"
2023-01-13T10:14:27.064Z	INFO	Loaded /root/app/trivy.yaml
2023-01-13T10:14:27.067Z	INFO	"--dependency-tree" only shows the dependents of vulnerable packages. Note that it is the reverse of the usual dependency tree, which shows the packages that depend on the vulnerable package. It supports "package-lock.json", "Cargo.lock" and OS packages. Please see the document for the detail.
2023-01-13T10:14:27.067Z	DEBUG	'--dependency-tree' enables '--list-all-pkgs'.
2023-01-13T10:14:27.067Z	DEBUG	Severities: ["MEDIUM" "HIGH" "CRITICAL"]
2023-01-13T10:14:27.081Z	DEBUG	cache dir:  /root/.cache/trivy
2023-01-13T10:14:27.081Z	DEBUG	There is no valid metadata file: unable to open a file: open /root/.cache/trivy/db/metadata.json: no such file or directory
2023-01-13T10:14:27.081Z	INFO	Need to update DB
2023-01-13T10:14:27.081Z	INFO	DB Repository: ghcr.io/aquasecurity/trivy-db
2023-01-13T10:14:27.081Z	INFO	Downloading DB...
2023-01-13T10:14:27.081Z	DEBUG	no metadata file
36.05 MiB / 36.05 MiB [------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 20.04 MiB p/s 2.0s
2023-01-13T10:14:30.216Z	DEBUG	Updating database metadata...
2023-01-13T10:14:30.216Z	DEBUG	DB Schema: 2, UpdatedAt: 2023-01-13 06:07:02.14805908 +0000 UTC, NextUpdate: 2023-01-13 12:07:02.14805848 +0000 UTC, DownloadedAt: 2023-01-13 10:14:30.216074864 +0000 UTC
2023-01-13T10:14:30.216Z	INFO	Vulnerability scanning is enabled
2023-01-13T10:14:30.216Z	DEBUG	Vulnerability type:  [os library]
2023-01-13T10:14:30.216Z	INFO	Misconfiguration scanning is enabled
2023-01-13T10:14:30.216Z	DEBUG	Walk the file tree rooted at '/root/app' in parallel
2023-01-13T10:14:30.241Z	DEBUG	Analysis error: unable to parse yarn.lock: failed to parse yarn.lock: failed to parse package patterns
2023-01-13T10:14:36.431Z	DEBUG	OS is not detected.
2023-01-13T10:14:36.431Z	DEBUG	Detected OS: unknown
2023-01-13T10:14:36.431Z	INFO	Number of language-specific files: 1
2023-01-13T10:14:36.431Z	INFO	Detecting pip vulnerabilities...
2023-01-13T10:14:36.431Z	DEBUG	Detecting library vulnerabilities, type: pip, path: requirements.txt
2023-01-13T10:14:36.431Z	INFO	Detected config files: 0

requirements.txt (pip)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

I've just tried `aquasec/trivy:0.35.0` & it seems to work fine.

```
docker run --rm -it \
  --volume="/opt/dev/academy:/root/app/" \
  aquasec/trivy:0.35.0 fs "/root/app" \
    --skip-dirs="node_modules" \
    --config="/root/app/trivy.yaml"
Unable to find image 'aquasec/trivy:0.35.0' locally
0.35.0: Pulling from aquasec/trivy
213ec9aee27d: Already exists 
7135c3aaaeef: Pull complete 
184bab12c89b: Pull complete 
b008d9a4945f: Pull complete 
Digest: sha256:9ff19e5b6cc17b5dd1e5e7ce46023d98ec03f80c4906024e7e23f5aad0606475
Status: Downloaded newer image for aquasec/trivy:0.35.0
2023-01-13T10:24:07.345Z	INFO	Loaded /root/app/trivy.yaml
2023-01-13T10:24:07.347Z	INFO	"--dependency-tree" only shows the dependents of vulnerable packages. Note that it is the reverse of the usual dependency tree, which shows the packages that depend on the vulnerable package. It supports "package-lock.json", "Cargo.lock" and OS packages. Please see the document for the detail.
2023-01-13T10:24:07.347Z	DEBUG	'--dependency-tree' enables '--list-all-pkgs'.
2023-01-13T10:24:07.347Z	DEBUG	Severities: ["MEDIUM" "HIGH" "CRITICAL"]
2023-01-13T10:24:07.361Z	DEBUG	cache dir:  /root/.cache/trivy
2023-01-13T10:24:07.361Z	DEBUG	There is no valid metadata file: unable to open a file: open /root/.cache/trivy/db/metadata.json: no such file or directory
2023-01-13T10:24:07.361Z	INFO	Need to update DB
2023-01-13T10:24:07.361Z	INFO	DB Repository: ghcr.io/aquasecurity/trivy-db
2023-01-13T10:24:07.361Z	INFO	Downloading DB...
2023-01-13T10:24:07.361Z	DEBUG	no metadata file
36.05 MiB / 36.05 MiB [------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 20.38 MiB p/s 2.0s
2023-01-13T10:24:10.379Z	DEBUG	Updating database metadata...
2023-01-13T10:24:10.379Z	DEBUG	DB Schema: 2, UpdatedAt: 2023-01-13 06:07:02.14805908 +0000 UTC, NextUpdate: 2023-01-13 12:07:02.14805848 +0000 UTC, DownloadedAt: 2023-01-13 10:24:10.379291764 +0000 UTC
2023-01-13T10:24:10.379Z	INFO	Vulnerability scanning is enabled
2023-01-13T10:24:10.379Z	DEBUG	Vulnerability type:  [os library]
2023-01-13T10:24:10.379Z	INFO	Misconfiguration scanning is enabled
2023-01-13T10:24:10.379Z	DEBUG	Walk the file tree rooted at '/root/app' in parallel
2023-01-13T10:24:16.567Z	DEBUG	OS is not detected.
2023-01-13T10:24:16.567Z	DEBUG	Detected OS: unknown
2023-01-13T10:24:16.567Z	INFO	Number of language-specific files: 2
2023-01-13T10:24:16.567Z	INFO	Detecting pip vulnerabilities...
2023-01-13T10:24:16.567Z	DEBUG	Detecting library vulnerabilities, type: pip, path: requirements.txt
2023-01-13T10:24:16.567Z	INFO	Detecting yarn vulnerabilities...
2023-01-13T10:24:16.567Z	DEBUG	Detecting library vulnerabilities, type: yarn, path: yarn.lock
2023-01-13T10:24:16.584Z	INFO	Detected config files: 0

requirements.txt (pip)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)


yarn.lock (yarn)

Total: 3 (MEDIUM: 1, HIGH: 2, CRITICAL: 0)

┌───────────┬────────────────┬──────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────┐
│  Library  │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                          Title                           │
├───────────┼────────────────┼──────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────┤
│ engine.io │ CVE-2022-41940 │ MEDIUM   │ 4.1.2             │ 6.2.1, 3.6.1  │ engine.io: Specially crafted HTTP request can trigger an │
│           │                │          │                   │               │ uncaught exception                                       │
│           │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-41940               │
├───────────┼────────────────┼──────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────┤
│ json5     │ CVE-2022-46175 │ HIGH     │ 1.0.1             │ 1.0.2, 2.2.2  │ json5: Prototype Pollution in JSON5 via Parse Method     │
│           │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-46175               │
│           │                │          ├───────────────────┤               │                                                          │
│           │                │          │ 2.2.1             │               │                                                          │
│           │                │          │                   │               │                                                          │
└───────────┴────────────────┴──────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────┘
make: *** [Makefile:66: audit] Error 1
```

--- a/go.mod
+++ b/go.mod
@@ -9,7 +9,7 @@ require (
 	github.com/alicebob/miniredis/v2 v2.23.0
 	github.com/aquasecurity/bolt-fixtures v0.0.0-20200903104109-d34e7f983986
 	github.com/aquasecurity/defsec v0.82.7-0.20230103191626-4f3187c78d6b
-	github.com/aquasecurity/go-dep-parser v0.0.0-20230105081339-fe9e63bf16bf
+	github.com/aquasecurity/go-dep-parser v0.0.0-20230115135733-3be7cb085121
 	github.com/aquasecurity/go-gem-version v0.0.0-20201115065557-8eed6fe000ce
 	github.com/aquasecurity/go-npm-version v0.0.0-20201110091526-0b796d180798
 	github.com/aquasecurity/go-pep440-version v0.0.0-20210121094942-22b2f8951d46
@@ -236,7 +236,7 @@ require (
 	github.com/hashicorp/errwrap v1.1.0 // indirect
 	github.com/hashicorp/go-cleanhttp v0.5.2 // indirect
 	github.com/hashicorp/go-multierror v1.1.1
-	github.com/hashicorp/go-retryablehttp v0.7.1 // indirect
+	github.com/hashicorp/go-retryablehttp v0.7.2 // indirect
 	github.com/hashicorp/go-safetemp v1.0.0 // indirect
 	github.com/hashicorp/go-uuid v1.0.3 // indirect
 	github.com/hashicorp/go-version v1.6.0 // indirect

